nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,346 @@
+"""
+Dictionary learning estimator: Perform a map learning algorithm based on
+component sparsity
```

Comment:
As far as I understand it, sparse PCA typcially imposes an L1 penalty term on the single voxels, whereas sparse dictionary learning typically imposes an L1 penalty term on the coefficients of the entire components of the dictionary, rather than the voxels of the space that is explained by the components.

Exactly for these aspect, I feel, that the distinction between DictLearning and SparsePCA should be explicit in both examples and nilearn base code.